File: Programing_Basics_with_JS/conditionalStatementsAdvancedMoreExercises/01.MatchTickets.js
function matchTickets(input) {
    let budget = Number(input[0]);
    let ticketType = input[1];
    let peopleInGroup = Number(input[2]);

    let transportPrice = 0;
    let vipTicketPrice = 499.99;
    let normalTicketPrice = 249.99;

    if (peopleInGroup <= 4) {
        transportPrice = budget * 0.75;
    } else if (peopleInGroup <= 9) {
        transportPrice = budget * 0.60;
    } else if (peopleInGroup <= 24) {
        transportPrice = budget * 0.50;
    } else if (peopleInGroup <= 49) {
        transportPrice = budget * 0.40;
    } else {
        transportPrice = budget * 0.25;
    }

    let budgetForTickets = budget - transportPrice;
    let allTicketsPrice = 0;

    if (ticketType === 'VIP') {
        allTicketsPrice = peopleInGroup * vipTicketPrice;
    } else {
        allTicketsPrice = peopleInGroup * normalTicketPrice;
    }

    let ticketPriceDiff = Math.abs(budgetForTickets - allTicketsPrice);

    if (budgetForTickets >= allTicketsPrice) {
        console.log(`Yes! You have ${ticketPriceDiff.toFixed(2)} leva left.`);
    } else {
        console.log(`Not enough money! You need ${ticketPriceDiff.toFixed(2)} leva.`);
    }
}

matchTickets(['1000', 'Normal', '1'])
matchTickets(['30000', 'VIP', '49'])

File: Programming_Basics_with_Python/conditional_statements_advanced_more_exercises/01._match_tickets.py
def tickets_budget(budget, category, people_count, vip_ticket_price, normal_ticket_price):
    transport_budget = 0
    all_tickets_price = 0

    if 1 <= people_count <= 4:
        transport_budget = budget * 0.75
    elif 5 <= people_count <= 9:
        transport_budget = budget * 0.6
    elif 10 <= people_count <= 24:
        transport_budget = budget * 0.5
    elif 25 <= people_count <= 49:
        transport_budget = budget * 0.4
    elif 50 <= people_count:
        transport_budget = budget * 0.25

    if category == 'VIP':
        all_tickets_price = people_count * vip_ticket_price
    elif category == 'Normal':
        all_tickets_price = people_count * normal_ticket_price

    budget_difference = budget - (transport_budget + all_tickets_price)

    if budget_difference >= 0:
        return f'Yes! You have {budget_difference:.2f} leva left.'
    else:
        return f'Not enough money! You need {abs(budget_difference):.2f} leva.'


budget = float(input())
category = input()
people_count = int(input())

vip_ticket_price = 499.99
normal_ticket_price = 249.99

print(tickets_budget(budget, category, people_count, vip_ticket_price, normal_ticket_price))